http-filter-example/http_filter.cc

```diff
@@ -1,18 +1,60 @@
 #include <string>
+#include <unordered_map>
+#include <sstream>
+#include <vector>
 
 #include "http_filter.h"
 
+#include "source/common/common/logger.h"
 #include "envoy/server/filter_config.h"
 
 namespace Envoy {
 namespace Http {
 
 HttpSampleDecoderFilterConfig::HttpSampleDecoderFilterConfig(
     const sample::Decoder& proto_config)
-    : key_(proto_config.key()), val_(proto_config.val()) {}
+    : key_(proto_config.key()), val_(proto_config.val()), extra_(stoi(proto_config.extra())) {}
 
 HttpSampleDecoderFilter::HttpSampleDecoderFilter(HttpSampleDecoderFilterConfigSharedPtr config)
-    : config_(config) {}
+    : config_(config) {
+  // populate list of valid operations, implementation will be changed in the next few commits
+  operations_.insert("set-header");
+  
+  const std::string header_config = headerValue();
+
+  // find the position of the first space character
+  const size_t spacePos = header_config.find(' ');
```

Comment:
I think this should check `std::string::npos` and `setError; return` if so. Otherwise, if the config is malformed and doesn't have a space, the `substr` below is going to throw an exception.

The fact that Envoy has an idiom to `setError` in constructors makes me think they're avoiding exceptions in constructors. I'm not 100% sure of that, but I see other extensions catching exceptions and returning, rather than throwing in constructors: https://github.com/envoyproxy/envoy/blob/c49f3925/source/extensions/tracers/xray/localized_sampling.cc#L97C1-L101